pojo/src/main/java/com/spotify/hamcrest/pojo/IsPojo.java

```diff
@@ -38,9 +47,10 @@
 public class IsPojo<A> extends TypeSafeDiagnosingMatcher<A> {
 
   private final Class<A> cls;
-  private final ImmutableMap<String, Matcher<?>> methodMatchers;
+  private final ImmutableMap<Function<A, ?>, Matcher<?>> methodMatchers;
```

Comment:
I do not think we want them to collide, otherwise following case will be impossible:
https://github.com/spotify/java-hamcrest/pull/19/files#diff-a284c930d1a406d7ffee257ebf1b9477R231

So in this case `Object.hashCode()` implementation is perfect.